src/NodeWrapper.ts

```diff
@@ -472,4 +474,86 @@ export default class NodeWrapper extends SelectableObject {
     this.nodeAcceptCircle.stroke(StateManager.colorScheme.nodeAcceptStrokeColor);
     this.nodeLabel.fill(StateManager.colorScheme.nodeLabelColor);
   }
+
+  public setErrorState(isError: boolean) {
+    this._isErrorState = isError;
+    this.updateAppearance();
+}
+
+
+private updateAppearance() {
+  // Remove existing error elements (if they exist) to avoid duplicates
+  this.nodeGroup.find('.errorIcon, .errorText').forEach((el) => el.destroy());
+
+  if (this._isErrorState) {
+      // Error state appearance
+      this.nodeBackground.fill('rgb(254, 226, 225)'); // Light red
+      this.nodeBackground.stroke('rgb(220, 38, 37)'); // Dark red
```

Comment:
It looks to me like if a node has errors, it will appear the same regardless of whether it is selected or not. We should allow the blue selection stroke to override the red error stroke (or perhaps be a second stroke around the border of the main one?) so that the user can always tell from looking at their diagram which nodes are selected.

Reply:
I've tried it. Now, I need your feedback on whether you like the visual appearance. If we want the blue stroke to remain visible even when the node is in an error state, it will look like this. This also means that if the error node isn't selected, it will have a black stroke inside.
<img width="116" alt="Screenshot 2024-11-14 at 09 33 03" src="https://github.com/user-attachments/assets/cfaa67ed-536b-4662-ae45-e0cf209a1d9b">
<img width="103" alt="Screenshot 2024-11-14 at 09 33 12" src="https://github.com/user-attachments/assets/7450fae3-ff52-4dec-b47b-8adfd8927dbb">

Reply:
Thanks for giving the double outline a try! Looking at it here, I'm having trouble seeing the blue outline at the same time as the red - they seem to blend together a bit for me. Perhaps a better solution would be to have the blue outline with no red outline, but keep the inside of the state tinted red as it was before?

I'll ask our capstone team what their thoughts on this are as well.

Reply:
I think this solution is best. You could also just have the state tinted red, and only give a red outline when its selected.

Reply:
I agree with Meorge on this. Having the regular blue outline with the filled in red color would be more consistent.

Reply:
<img width="479" alt="image" src="https://github.com/user-attachments/assets/fcbc4e02-b7ce-49f8-8a4b-c07ce066c0d3">

Do you mean something like this? Or when the node is in error state and is selected there should be no "X" sign?

Reply:
Yeah thats pretty much how I imagined it, looks great!

Reply:
That looks good to me as well!

Reply:
Updated version looks good on my end!